packages/rocketchat-api/server/v1/channels.js

```diff
@@ -593,16 +581,15 @@ RocketChat.API.v1.addRoute('channels.messages', { authRequired: true }, {
 	get() {
 		const findResult = findChannelByIdOrName({
 			params: this.requestParams(),
-			checkedArchived: false,
-			returnUsernames: true
+			checkedArchived: false
 		});
 		const { offset, count } = this.getPaginationItems();
 		const { sort, fields, query } = this.parseJsonQuery();
 
 		const ourQuery = Object.assign({}, query, { rid: findResult._id });
 
 		//Special check for the permissions
-		if (RocketChat.authz.hasPermission(this.userId, 'view-joined-room') && !findResult.usernames.includes(this.user.username)) {
+		if (RocketChat.authz.hasPermission(this.userId, 'view-joined-room') && !RocketChat.models.Subscriptions.findOneByRoomIdAndUserId(findResult._id, this.userId, {fields: {_id: 1}})) {
```

Comment:
do you think this way will be worst ?
```javascript
		const ids = online.map((result, {_id}) => _id)

		const online = RocketChat.models.Subscriptions.find({ rid: root._id, _id: { $in: ids } }, { fields: { _id: 1, username: 1 } });
		return RocketChat.API.v1.success({ online });
```

Reply:
@ggazzo did you comment at the right place?

Reply:
maybe :p